Mathlib/RingTheory/CotangentLocalizationAway.lean

```diff
@@ -0,0 +1,145 @@
+/-
+Copyright (c) 2025 Christian Merten. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Christian Merten
+-/
+import Mathlib.RingTheory.Presentation
+
+/-!
+# Cotangent and localization away
+
+Let `R → S → T` be algebras such that `T` is the localization of `S` away from one
+element, where `S` is generated over `R` by `P` with kernel `I` and `Q` is the
+canonical `S`-presentation of `T`. Denote by `J` the kernel of the composition
+`R[X,Y] → T`.
+
+## Main results
+
+- `Algebra.Generators.liftBaseChange_injective`:
+  `T ⊗[S] (I/I²) → J/J²` is injective if `T` is the localization of `S` away from an element.
+
+-/
+
+open TensorProduct MvPolynomial
+
+namespace Algebra.Generators
+
+variable {R S T : Type*} [CommRing R] [CommRing S] [Algebra R S]
+  [CommRing T] [Algebra R T] [Algebra S T] [IsScalarTower R S T]
+variable (g : S) [IsLocalization.Away g T] (P : Generators R S)
+
+-- Allow seeing through the `vars` field of `Generators`.
+-- TODO: eliminate this by refactoring `Generators` and `Presentation`,
+-- either unbundling the `vars` field or making the field globally reducible in constructions.
```

Comment:
Is this something we surely want? In this case can you add a `TODO` where is should be done?

Reply:
The refactor mentioned is stuck on https://leanprover.zulipchat.com/#narrow/channel/287929-mathlib4/topic/JacobiZariski.20is.20slow.2E. But I don't think we should halt all progress for that.

Reply:
I agree, I just wanted a TODO, but if there is a Zulip discussion even better.